examples/tutorials/manage-ntp.markdown

```diff
@@ -0,0 +1,741 @@
+server {{{.}}} iburst
+{{/servers}}
+includefile /etc/ntp/crypto/pw
+keys /etc/ntp/keys
+";
+
+```
+
+A few new variables are defined. The variables `ntp_package_name`, `config_file`, `driftfile`, `servers`, and `config_template_string` are defined under the `linux` context (so only linux hosts will define them). `config_file` is the path to the ntp configuration file, `driftfile` and `servers` are both variables that will be used when rendering the configuration file and `config_template_string` is the template that will be used to render the configuration file. While both `driftfile** and `servers** are set the same for all linux hosts, those variables could easily be set to different values under different contexts.
+
+#### files
+
+Now let's walk through the files promise in detail.
+
+```cf3
+   files:
+    "$(config_file)"
+      create                => "true",
+      handle                => "ntp_files_conf",
+      perms                 => mog( "644", "root", "root" ),
+      template_method       => "inline_mustache",
+      edit_template_string  => "$(config_template_string)",
+      template_data         => mergedata( '{ "driftfile": "$(driftfile)", "servers": servers}' ),
+      classes               => results( "bundle", "ntp_config" );
+```
+
+The promiser here is referenced by the `config_file` variable. In this case, it is the configuration file for the NTP service. There are a number of additional attributes that describe this promise.
+
+##### create
+
+```cf3
+      create                => "true",
+```
+
+Valid values for this attribute are `true` or `false` to instruct the agent whether or not to create the file. In other words, the file must exist. If it does not exist, it will be created.
+
+##### perms
+
+```cf3
+      perms                 => mog( "644", "root", "root" ),
+```
+
+This attribute sets the permissions and ownership of the file. `mog()` is function in the CFEngine standard library that sets the `mode`, `owner`, and `group` of the file. In this example, the permissions for the NTP configuration file is set to `644` with owner and group both assigned to `root`.
+
+##### handle
+
+```cf3
+      handle                => "ntp_files_conf",
+```
+
+This is a descriptive handle that will uniquely identify the promise within this policy. The [policy style guide][Policy Style Guide#promise handles] recommends a naming scheme for the handles e.g. `bundle_name_promise_type_class_restriction_promiser`.
+
+##### classes
+
+```cf3
+      classes               => results( "bundle", "ntp_config" );
+```
+The classes attribute here indicates that every time this promise is triggered, whether the file is created, or the content of the file is updated, the class `restart_ntpd` will be set. Recall that this class was also set in the processes promise when the agent determines that the NTP process is not running. Setting this class activates the commands promise which will proactively issue a command to restart the service.
```

Comment:
`restart_ntpd` will not be set, `ntp_config_repaired` will.

Reply:
Also no processes and commands promises.

Reply:
ACK